test/test_binary_ufuncs.py

```diff
@@ -2572,6 +2573,56 @@ def test_x(sizes, dim, x, device):
             test_x((2, 3), 1, [1.0, 2.0], device)
             test_x((2, 3), 1, [1.0, 2.0, 3.0, 4.0], device)
 
+    @skipIf(not TEST_SCIPY, "Scipy required for the test.")
+    def test_cumulative_trapezoid(self, device):
+
+        import scipy.integrate
+
+        def test_dx(sizes, dim, dx, device):
+            t = torch.randn(sizes, device=device)
+            y = t.cpu().numpy()
+            actual = torch.cumulative_trapezoid(t, dx=dx, dim=dim)
+            expected = scipy.integrate.cumulative_trapezoid(t.cpu().numpy(), dx=dx, axis=dim)
+            self.assertEqual(expected.shape, actual.shape)
+            self.assertEqual(expected, actual, exact_dtype=False)
+
+        def test_x(sizes, dim, x, device):
+            t = torch.randn(sizes, device=device)
+            actual = torch.cumulative_trapezoid(t, x=torch.tensor(x, device=device), dim=dim)
+            expected = scipy.integrate.cumulative_trapezoid(t.cpu().numpy(), x=x, axis=dim)
+            self.assertEqual(expected.shape, actual.shape)
+            self.assertEqual(expected, actual.cpu(), exact_dtype=False)
+
+        test_dx((2,), -1, 1, device)
+        test_dx((3, 3), -1, 1, device)
+        test_dx((4, 2), 0, 1, device)
+        test_dx((2, 3, 4), 1, 1, device)
+        test_dx((10, 2), 0, 0.1, device)
+        test_dx((1, 10), 0, 2.3, device)
+        test_dx((0, 2), 0, 1.0, device)
+        test_dx((0, 2), 1, 1.0, device)
+
+        test_x((2,), -1, [100, 50], device)
+        test_x((4, 2), 0, [2, 3, 4, 5], device)
+        test_x((2, 3, 4), 1, [1.0, 2.0, 3.0], device)
+        test_x((10, 2), 0, [2.0, 3.0, 4.0, 7.0, 11.0, 14.0, 22.0, 26.0, 26.1, 30.3], device)
+        test_x((1, 10), 0, [1.0], device)
+        test_x((0, 2), 1, [1, 2], device)
+
+        # test_x((0, 2), 0, [], device) # Scipy failing because x == [], but our version returns empty
+
+        with self.assertRaisesRegex(
```

Comment:
What about a test that verifies specifying both `x` and `dx` simultaneously is an error? And one passing `dx` as a complex value?

Reply:
Added both.